File: src/main/java/io/stouder/adonis/activity/InitRoutesActivity.java
package io.stouder.adonis.activity;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.startup.ProjectActivity;
import io.stouder.adonis.notifier.AdonisRouteUpdateNotifier;
import io.stouder.adonis.service.AdonisRouteService;
import kotlin.Unit;
import kotlin.coroutines.Continuation;
import org.jetbrains.annotations.NotNull;

public class InitRoutesActivity implements ProjectActivity {

    @Override
    public Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
        AdonisRouteUpdateNotifier publisher = project.getMessageBus().syncPublisher(AdonisRouteUpdateNotifier.ADONIS_ROUTES_UPDATE_TOPIC);
        AdonisRouteService.getInstance(project).fetchRoutes(publisher::routes);
        return Unit.INSTANCE;
    }
}

File: src/main/java/io/stouder/adonis/activity/InitRoutesActivity.kt
package io.stouder.adonis.activity

import com.intellij.openapi.project.Project
import com.intellij.openapi.startup.ProjectActivity
import io.stouder.adonis.notifier.AdonisRouteUpdateNotifier
import io.stouder.adonis.service.AdonisRouteService

class InitRoutesActivity : ProjectActivity {
    override suspend fun execute(project: Project) {
        val publisher = project.messageBus.syncPublisher(AdonisRouteUpdateNotifier.ADONIS_ROUTES_UPDATE_TOPIC)
        AdonisRouteService.getInstance(project).fetchRoutes(publisher::routes)
    }
}